toolkit/tools/graphpkgfetcher/graphpkgfetcher.go

```diff
@@ -56,29 +56,65 @@ func main() {
 	kingpin.MustParse(app.Parse(os.Args[1:]))
 	logger.InitBestEffort(*logFile, *logLevel)
 
+	err := fetchPackages()
+	if err != nil {
+		logger.Log.Fatalf("Failed to fetch packages. Error: %s", err)
+	}
+}
+
+func fetchPackages() (err error) {
 	dependencyGraph, err := pkggraph.ReadDOTGraphFile(*inputGraph)
 	if err != nil {
-		logger.Log.Panicf("Failed to read graph to file. Error: %s", err)
+		err = fmt.Errorf("failed to read graph to file: %w", err)
+		return
 	}
 
 	toolchainPackages, err := schedulerutils.ReadReservedFilesList(*toolchainManifest)
 	if err != nil {
-		logger.Log.Fatalf("unable to read toolchain manifest file '%s': %s", *toolchainManifest, err)
+		err = fmt.Errorf("unable to read toolchain manifest file '%s': %w", *toolchainManifest, err)
+		return
 	}
 
-	if hasUnresolvedNodes(dependencyGraph) {
-		err = resolveGraphNodes(dependencyGraph, *inputSummaryFile, *outputSummaryFile, toolchainPackages, *disableUpstreamRepos, *stopOnFailure)
+	var cloner *rpmrepocloner.RpmRepoCloner
+	hasUnresolvedNodes := hasUnresolvedNodes(dependencyGraph)
+	if hasUnresolvedNodes {
+		// Create the worker environment
+		cloner, err = prepRpmCloner(*outDir, *disableUpstreamRepos)
 		if err != nil {
```

Comment:
Let's keep explicit `var` definitions at the beginning of the function - it's what we tend to stick to across our code. Similar to `const`. Also, consider this alternative:
```suggestion
	hasUnresolvedNodes := hasUnresolvedNodes(dependencyGraph)
	if hasUnresolvedNodes {
		// Create the worker environment
		cloner, err := prepRpmCloner(*outDir, *disableUpstreamRepos)
		if err != nil {
		err = ...
		return err
	}
```

Note: one must here explicitly use `return err` instead of an empty `return`. I couldn't make the suggestion directly "mergeable" into your code - GitHub doesn't handle code suggestions for sections containing removed lines.:/

Reply:
Addressed